src/test/java/com/autosale/shop/controller/UserControllerTest.java

```diff
@@ -0,0 +1,118 @@
+    private final ObjectMapper mapper = new ObjectMapper();
+
+
+    List<User> users = new ArrayList<>();
+
+    @BeforeEach
+    void setUp() {
+        Random rand = new Random();
+        for (int i = 0; i < 3; i++) {
+            users.add(new User(rand.nextInt(100), RandomString.make(50), RandomString.make(25), rand.nextBoolean() ? UserRole.USER : UserRole.ADMIN));
+        }
+    }
+
+    @AfterEach
+    void finale() {
+        users.clear();
+    }
+
+    @Test
+    void findAll() throws Exception {
+
+        when(service.findAll()).thenReturn(users);
+        MvcResult result = mock.perform(MockMvcRequestBuilders.get("/users"))
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON)).andReturn();
+        assertThat(new String(result.getResponse().getContentAsByteArray(), UTF_8), is(equalTo(mapper.writeValueAsString(users))));
+    }
+
+    @Test
+    void create() throws Exception {
+        User user = users.get(0);
+        when(service.create(user)).thenReturn(user.getId());
+        MvcResult result = mock.perform(post("/users")
+                        .content(mapper.writeValueAsString(user))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        assert (result.getRequest().getContentAsByteArray() != null);
+        assertThat(HttpStatus.OK.value(), is(equalTo(result.getResponse().getStatus())));
+        assertThat(new String(result.getRequest().getContentAsByteArray(), UTF_8), is(equalTo(mapper.writeValueAsString(user))));
+
+    }
+
+    @Test
+    void findById() throws Exception {
+        User user = users.get(0);
+        when(service.findById(user.getId())).thenReturn(user);
+        MvcResult result = mock.perform(get("/users/"+user.getId())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        assertThat(HttpStatus.OK.value(), is(equalTo( result.getResponse().getStatus())));
+        assertThat(new String(result.getResponse().getContentAsByteArray(),UTF_8),is(equalTo(mapper.writeValueAsString(user))));
+    }
+
+    @Test
+    void delete() throws Exception {
+        User user = users.get(0);
+        when(service.delete(user.getId())).thenReturn(1);
+        MvcResult result = mock.perform(MockMvcRequestBuilders.delete("/users/" + user.getId())
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        assertThat(HttpStatus.OK.value(), is(equalTo(result.getResponse().getStatus())));
+        assertThat(new String(result.getResponse().getContentAsByteArray(), UTF_8), is(equalTo("1")));
+    }
+
+    @Test
+    void update() throws Exception {
+        User user = users.get(0);
+        MvcResult result = mock.perform(put("/users")
+                        .content(mapper.writeValueAsString(user))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
```

Comment:
how?

Reply:
mock.perform(put("/users")
                        .content(mapper.writeValueAsString(user))
                        .contentType(MediaType.APPLICATION_JSON))
                .andReturn()
                .getResponse();

Reply:
Maybe I will use it in the future, but now this option does not suit me because in this test I need both a request (for check data transfer) and a response (for check httpStatus)